README
NAME
    Paludis::ResumeState::Serialization - Work with resume-files generated
    by Paludis

VERSION
    version 0.01000409

SYNOPSIS
        use Paludis::ResumeState::Serialization;

        open my $fh, '<' , '/resumefile' or die;

        my $objects = Paludis::ResumeState::Serialization->deserialize({
            content => ( do { local $/ = undef;  scalar <$fh> } ),
            format => 'basic'
        });

        my $content = Paludis::ResumeState::Serialization->serialize({
            data => $object,
            format => 'basic'
        });

        # $content should == contents of resumefile.

    This class is just really a proxy serialization interface for a few of
    the varying back-ends.

    Currently only the 'basic' back-end exists, which provides basic, but
    consistent serialization support.

  FormatNames
   basic
    Defers serialization to "::Basic"

METHODS
  deserialize
        my $object = ::Serialization->deserialize({
            content => $string
            format  => FormatName
        });

    See "FormatNames"

  serialize
        my $string = ::Serialization->serialize({
            data => $object,
            format => FormatName
        });

    See "FormatNames"

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2011 by Kent Fredric <kentnl@cpan.org>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for {{$dist->name}}

{{$NEXT}}

0.01000410 2013-11-09T18:06:41Z
 [00 Bugfix]
 - Avoid Regexp::Grammars bug on >5.18 which inhibit calling local
   functions, by fully qualifiying the relevant functions.
 - Avoid bogus test failures on >5.18 due to tests being inherently
   key-order dependent

 [Dependencies::Added / develop requires]
 - Dist::Zilla::Plugin::Author::KENTNL::MinimumPerl
 - Dist::Zilla::Plugin::Authority 1.006
 - Dist::Zilla::Plugin::AutoPrereqs
 - Dist::Zilla::Plugin::CheckExtraTests
 - Dist::Zilla::Plugin::ConfirmRelease
 - Dist::Zilla::Plugin::EOLTests
 - Dist::Zilla::Plugin::Git::Check
 - Dist::Zilla::Plugin::Git::Commit
 - Dist::Zilla::Plugin::Git::CommitBuild
 - Dist::Zilla::Plugin::Git::GatherDir
 - Dist::Zilla::Plugin::Git::NextVersion
 - Dist::Zilla::Plugin::Git::Tag
 - Dist::Zilla::Plugin::GithubMeta
 - Dist::Zilla::Plugin::License
 - Dist::Zilla::Plugin::Manifest
 - Dist::Zilla::Plugin::ManifestSkip
 - Dist::Zilla::Plugin::MetaConfig
 - Dist::Zilla::Plugin::MetaData::BuiltWith
 - Dist::Zilla::Plugin::MetaJSON
 - Dist::Zilla::Plugin::MetaProvides::Package 1.14000001
 - Dist::Zilla::Plugin::MetaTests
 - Dist::Zilla::Plugin::MetaYAML
 - Dist::Zilla::Plugin::ModuleBuild
 - Dist::Zilla::Plugin::NextRelease
 - Dist::Zilla::Plugin::PkgVersion
 - Dist::Zilla::Plugin::PodCoverageTests
 - Dist::Zilla::Plugin::PodSyntaxTests
 - Dist::Zilla::Plugin::PodWeaver
 - Dist::Zilla::Plugin::Prereqs
 - Dist::Zilla::Plugin::Prereqs::MatchInstalled
 - Dist::Zilla::Plugin::PruneCruft
 - Dist::Zilla::Plugin::ReadmeAnyFromPod
 - Dist::Zilla::Plugin::ReadmeFromPod
 - Dist::Zilla::Plugin::ReportVersions::Tiny
 - Dist::Zilla::Plugin::Test::CPAN::Changes
 - Dist::Zilla::Plugin::Test::Compile
 - Dist::Zilla::Plugin::Test::Kwalitee
 - Dist::Zilla::Plugin::Test::MinimumVersion
 - Dist::Zilla::Plugin::Test::Perl::Critic
 - Dist::Zilla::Plugin::TestRelease
 - Dist::Zilla::Plugin::Twitter
 - Dist::Zilla::Plugin::UploadToCPAN
 - Dist::Zilla::PluginBundle::Author::KENTNL 2.001000
 - Pod::Coverage::TrustPod
 - Test::CPAN::Changes 0.19
 - Test::CPAN::Meta
 - Test::Kwalitee 1.12
 - Test::Pod 1.41
 - Test::Pod::Coverage 1.08
 - version 0.9901

 [Dependencies::Added / develop suggests]
 - Dist::Zilla::PluginBundle::Author::KENTNL::Lite v1.3.0

 [Dependencies::Added / runtime requires]
 - perl 5.010
 - strict
 - warnings

 [Dependencies::Added / test requires]
 - File::Spec
 - IO::Handle
 - IPC::Open3

 [Dependencies::Changed / build requires]
 - Module::Build 0.3601 → 0.4008

 [Dependencies::Changed / configure requires]
 - Module::Build 0.3601 → 0.4008

 [Dependencies::Changed / test requires]
 - Test::More 0.96 → 1.001002

 [Dependencies::Removed / develop recommends]
 - Dist::Zilla::PluginBundle::KENTNL::Lite 0.01009803

 [Dependencies::Removed / develop requires]
 - Dist::Zilla::PluginBundle::KENTNL::Lite

 [Dependencies::Removed / develop suggests]
 - Dist::Zilla::PluginBundle::KENTNL 0.01017119

 [Dependencies::Removed / test requires]
 - File::Find
 - File::Temp

 [Documentation]
 - Update LICENSE copyright year
 - Reindent LICENSE
 - Update LICENSE address
 - Use utf-8 everywhere
 - Ship README.mkdn on source tree

 [Internals]
 - $AUTHORITY in all modules
 - $VERSION lacks BEGIN

 [Meta]
 - Add x_Authority
 - Issues to Github issues
 - Website URL unbroken

 [Packaging]
 - ship .perltidyrc
 - use recent test_requires in Module::Build

 [Tests]
 - superflouous t/ tests moved to xt/
 - compile test regenerated.
 - report versions test regenerated

0.01000409 2011-01-01T09:42:23Z
 - First version.



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4008"
  },
  "configure_requires" => {
    "Module::Build" => "0.4008"
  },
  "dist_abstract" => "Work with resume-files generated by Paludis",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>"
  ],
  "dist_name" => "Paludis-ResumeState-Serialization",
  "dist_version" => "0.01000410",
  "license" => "perl",
  "module_name" => "Paludis::ResumeState::Serialization",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Class::Load" => "0.06",
    "English" => 0,
    "Params::Util" => 0,
    "Regexp::Grammars" => 0,
    "Regexp::Grammars::Common::String" => 0,
    "autovivification" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IO::Uncompress::Gunzip" => 0,
    "IPC::Open3" => 0,
    "Scalar::Util" => 0,
    "Test::More" => "1.001002"
  }
);


my %fallback_build_requires = (
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IO::Uncompress::Gunzip" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.4008",
  "Scalar::Util" => 0,
  "Test::More" => "1.001002"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
